### ISSUE TITLE
Question about using monotonic AvSGD

### ISSUE THREAD

**Issue by Ruowei94** [salesforce/awd-lstm-lm#34]
Hi,

I have some questions about the "Regularizing and Optimizing LSTM Languaguage models" paper. In the second line of the first paragraph in page 10 of the paper, you mentioned "using a monotonic criterion instead also hampered performance." I am not sure about what do you mean by "a monotonic criterion"? Do you mean by using AvSGD once the validation metric fails to improve? 

In addition, I am also confused about the "dropouti" flag uou used in main.py script in line 39. The help argument said "dropout for input embedding layers (0 = no dropout)", is "dropouti" the dropout applied to the input layer which is the before the embedding layer?
 
Thank you so much.

**Comment by keskarnitish** [comment]
> Do you mean by using AvSGD once the validation metric fails to improve?

Correct. That is what we meant. 

>  I am also confused about the "dropouti" flag uou used in main.py script in line 39.

For "dropouti" it might be useful to see how it is used in "model.py".

For applying dropout to the embeddings, there are two relevant hyper parameters:
"dropoute" - replace a word's vector with all zeros with probability "dropoute" (i.e. it's totally zeroed)
"dropouti" - for the word vector resulting from the above step, apply normal dropout to the representation (i.e. maybe 10% of the values in the word vector end up zeroed, but the rest remain the same)

This acts for two reasons:
"dropoute" prevents overfitting to exact sequences that the model may see in training
"dropouti" prevents the model from overfitting to the exact word representation
